### ISSUE TITLE
horovod hang with NCCL when ran mnist_advanced example

### ISSUE THREAD

**Issue by CDCrystal** [horovod/horovod#1221]
**Environment:**（the example/keras_mnist_advanced.py）
1. Framework: Keras
3. Horovod version:0.16.4
4. MPI version:OpenMPI 4.0.0
5. CUDA version:cuda10.0
6. NCCL version:2.4.7-1+cuda10.0
7. Python version:2.7
8. OS and version:Ubuntu 18.04+CentOS7 (both in Docker)
9. GCC version:4.8

I got the NCCL INFO NET/Plugin : No plugin found (libnccl-net.so). ERROR and freezed in Epoch 1/24. I‘ve checked the other issues but I'm still confused. 

GPU-Util was 0% but GPU memory was occupied.

![image](https://user-images.githubusercontent.com/43447571/61174163-293a2300-a5cf-11e9-9914-9695dec356f2.png)

**Comment by tgaddair** [comment]
Hey @CDCrystal, the NCCL message you're seeing can be safely ignored.  I suspect the issue has something to do with the MPI configuration.  You mentioned you're using both Ubuntu and CentOS.  Can you explain a bit more about your setup?  How many instances are you using, how many GPUs per instance?  What does your horovodrun/mpirun command look like?

**Comment by CDCrystal** [comment]
Hi, @tgaddair , I did have some configuration error with MPI that I forgot to set the corresponding network interfaces. I've rewrite the provided Dockerfile to adapt the CentOS 7 which has a low version of g++, but I missed the Dockerfile since the server was reinstalled the Ubuntu 16.04. I’ve used 2 machines each with one GPU.  My mpirun command is following:
![image](https://user-images.githubusercontent.com/43447571/61808460-03820900-ae6e-11e9-9872-eba70f1d1a48.png)
By reading the horovod/docs/mpirun.rst guide, I targeted the problem. (https://github.com/horovod/horovod/blob/master/docs/mpirun.rst)
Thanks for your attention.
